cmd/crank/install.go

```diff
@@ -120,6 +129,30 @@ func (c *installConfigCmd) Run(k *kong.Context, logger logging.Logger) error {
 		logger.Debug("Failed to create configuration", "error", warnIfNotFound(err))
 		return errors.Wrap(warnIfNotFound(err), "cannot create configuration")
 	}
+	if err = waitUntil(waitDuration, func() error {
```

Comment:
Does this just wait for the whole duration in the happy path? I believe we should be stopping the wait as soon as our condition (package is healthy) is met. I recommend using something like [wait.PollImmediate](https://github.com/kubernetes/apimachinery/blob/aec8116c445f5fa9d277e64ca432d978afae4ae3/pkg/util/wait/wait.go#L508) :+1:

Reply:
No, it actually returns in case of the happy path. If you take a look at this `waitUntil` function ([here](https://github.com/crossplane/crossplane/pull/2345/files#diff-373f6bfdec661e4a082de6049df5e7c0da1ae221df32c76cdbe61e7d6463f5e9R272)) if the `try` function returns `nil` then it returns immediately. For each try method, we return nil as soon as the resource is ready([here](https://github.com/crossplane/crossplane/pull/2345/files#diff-373f6bfdec661e4a082de6049df5e7c0da1ae221df32c76cdbe61e7d6463f5e9R243))

I believe this does kind of the same thing as `wait.PollImmediate` but we can replace this with `PollImmediate` which will give us a standardized approach.

Reply:
Ah I see, we are returning an error while condition is not yet met. I prefer `wait.PollImmediate` because we return a `(bool, error)` in the condition function, meaning we have the ability to:
1. Continue waiting when condition is not met: `(false, nil)`
2. Exit early when we encounter an error that is not related to the package not being healthy: `(false, err)`
3. Exit early when the package is healthy: `(true, nil)`

Reply:
Right, it is a more flexible pattern and allows early exit in case of unrelated errors. I have made the above change.